### PR TITLE
Add plan-variant for bintray-publish

### PR DESCRIPTION
**Pull request by smacfarlane**
This refactors bintray-publish so that we can build x86-64-linux-kernel2 variants.

This now also includes a number of relative path changes in other components. These changes are necessary to ensure we can build all of the components in a release, many of which come before bintray-publish in the release process.
Ref #5698 
Signed-off-by: Scott Macfarlane <macfarlane.scott@gmail.com>

### CODE REVIEW

**Comment by thesentinels** [comment]
Thanks for the pull request! Here is what will happen next:
 1. Your PR will be reviewed by the maintainers
 2. If everything looks good, one of them will `approve` it, and your PR will be merged.

Thank you for contributing!

**Comment by raskchanky** [comment]
The merge conflicts are likely my fault due to the changes I made yesterday in order to get appveyor tests passing so I could merge my PR.  Sorry 😞

**Comment by christophermaier** [comment]
@smacfarlane why are there so many file renames in things unrelated to the bintray package?

**Comment by smacfarlane** [comment]
It looks like some things were inadvertently moved as part of the fixes to get appveyor build working (which is totally on me, I misread what appveyor was telling me on the initial PR)  

Running through the "fake release" process revealed that studio among others could no longer find their source files to copy to the  artifact.  These extra changes are moving the bin directories back up one level to their original path.  

Ex: https://buildkite.com/chef/habitat-sh-habitat-master-release/builds/1462

File: components/studio/habitat/plan.sh [hunk 1]
```diff
@@ -46,21 +45,3 @@ do_install() {
     "$pkg_prefix"/libexec/"$hab_dir"/bin/hab
   ln -sv "$hab_dir"/bin/hab "$pkg_prefix"/libexec/hab
 }
-
-# Turn the remaining default phases into no-ops
```
Comment (baumanj) [inline components/studio/habitat/plan.sh]: It's not clear to me why this is no longer needed. Don't we have default implementations of these actions? If they're not overridden, they won't be no-ops, right?

Comment (smacfarlane) [reply components/studio/habitat/plan.sh]: That's a great thing to call out.  It used to be the case that `pkg_source` was "required" ( this was  two years ago so my memory is a bit fuzzy on what required actually meant ), which meant having to override the download/unpack/verify steps as there was nothing to actually do. 

Now you are able to omit the `pkg_source` and plan-build will automatically skip over those steps so we no longer need to no-op them.   

I'm not entirely sure on the `do_prepare` no-op, that may be an even older leftover from early iterations on plan-build, but the default for it is currently a no-op so we can safely omit it as well.

**Pull request merged**